fenix/app/src/main/java/org/mozilla/fenix/ext/FragmentActivity.kt

```diff
@@ -0,0 +1,181 @@
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.storage.HistoryMetadataKey
+import mozilla.components.support.ktx.kotlin.isUrl
+import mozilla.components.support.ktx.kotlin.toNormalizedUrl
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+
+/**
+ * Navigates to the browser fragment and loads a URL or performs a search (depending on the
+ * value of [searchTermOrURL]).
+ *
+ * @param navController The [NavController] to use for navigation.
+ * @param searchTermOrURL The entered search term to search or URL to be loaded.
+ * @param newTab Whether or not to load the URL in a new tab.
+ * @param from The [BrowserDirection] to indicate which fragment the browser is being opened from.
+ * @param browsingMode The tab's current [BrowsingMode].
+ * @param customTabSessionId Optional custom tab session ID if navigating from a custom tab.
+ * @param engine Optional [SearchEngine] to use when performing a search.
+ * @param forceSearch Whether or not to force performing a search.
+ * @param flags Flags that will be used when loading the URL (not applied to searches).
+ * @param requestDesktopMode Whether or not to request the desktop mode for the session.
+ * @param historyMetadata The [HistoryMetadataKey] of the new tab in case this tab was opened from history.
+ * @param additionalHeaders The extra headers to use when loading the URL.
+ */
+fun FragmentActivity.openToBrowserAndLoad(
+    navController: NavController,
+    searchTermOrURL: String,
+    newTab: Boolean,
+    from: BrowserDirection,
+    browsingMode: BrowsingMode = BrowsingMode.Normal,
+    customTabSessionId: String? = null,
+    engine: SearchEngine? = null,
+    forceSearch: Boolean = false,
+    flags: EngineSession.LoadUrlFlags = EngineSession.LoadUrlFlags.none(),
+    requestDesktopMode: Boolean = false,
+    historyMetadata: HistoryMetadataKey? = null,
+    additionalHeaders: Map<String, String>? = null,
+) {
+    openToBrowser(navController = navController, from = from, customTabSessionId = customTabSessionId)
+    load(
+        searchTermOrURL = searchTermOrURL,
+        newTab = newTab,
+        engine = engine,
+        forceSearch = forceSearch,
+        flags = flags,
+        browsingMode = browsingMode,
+        requestDesktopMode = requestDesktopMode,
+        historyMetadata = historyMetadata,
+        additionalHeaders = additionalHeaders,
+    )
+}
+
+/**
+ * Navigates to the browser fragment.
+ *
+ * @param navController The [NavController] to use for navigation.
+ * @param from The [BrowserDirection] to indicate which fragment the browser is being opened from.
+ * @param customTabSessionId Optional custom tab session ID if navigating from a custom tab.
+ */
+fun FragmentActivity.openToBrowser(
+    navController: NavController,
```

Comment:
navController is a new param that is passed in by the caller

Reply:
Is the `navController` ever different than the one attached to the activity? If so, why?